src/data/list/sort.lean

```diff
@@ -73,13 +73,31 @@ end
 
 @[simp] theorem sorted_singleton (a : α) : sorted r [a] := pairwise_singleton _ _
 
-lemma nth_le_of_sorted_of_le [is_refl α r] {l : list α}
+lemma sorted.rel_nth_le_of_lt {l : list α}
+  (h : l.sorted r) {a b : ℕ} {ha : a < l.length} {hb : b < l.length} (hab : a < b) :
```

Comment:
```suggestion
  (h : l.sorted r) {a b : ℕ} (ha : a < l.length) (hb : b < l.length) (hab : a < b) :
```
Since they can't be inferred, and the only way to use this lemma is via apply on a properly shaped goal.

Reply:
Right, but at the point you use this lemma those metavariables would be filled in by the rewrite / apply, so I can't envisage a used where we wouldn't just pass underscores. Case in point - not a single call site of this lemma passes arguments today.

Reply:
We have a style guide about this: they should be explicit unless they can be inferred from the *hypotheses*. The usage context can vary, and it's a lot more annoying to make implicit args explicit than vice versa.